src/components/BannerDialogComponent.vue

```diff
@@ -0,0 +1,216 @@
+// If Banner is undefined we are creating, otherwise editing.
+const banner = defineModel<BannerResponse | undefined>('banner');
+
+
+// Banner schema and validation
+// TODO: Actual validation and errors
+const bannerSchema = toTypedSchema(
+    yup.object({
+        'Name': yup.string().required(),
+        'Duration': yup.number().required().moreThan(0),
+        'Start date': yup.date().required(),
+        'End date': yup.date().required().min(new Date()).min(
+            yup.ref('Start date'),
+            "End date can't be before start date"
+        ),
+        'Active': yup.boolean().required(),
+    })
+);
+
+const {
+    errors,
+    defineField,
+    setFieldValue,
+    handleSubmit,
+    resetForm
+} = useForm({
+    validationSchema: bannerSchema
+});
+
+const [ name ] = defineField('Name');
+const [ duration ] = defineField('Duration');
+const [ startDate ] = defineField('Start date');
+const [ endDate ] = defineField('End date');
+const [ active ] = defineField('Active');
+
+// Reset values everytime dialog is opened.
+watch(visible, () => {
+    if(visible.value) {
+        resetValues();
+    }
+});
+
+function resetValues() {
+    resetForm();
+    setFieldValue('Name', banner.value?.name, false);
+    setFieldValue('Duration', banner.value && banner.value?.duration / 1000, false);
+    setFieldValue('Start date', banner.value && new Date(banner.value?.startDate), false);
+    setFieldValue('End date', banner.value && new Date(banner.value?.endDate), false);
+    setFieldValue('Active', banner.value?.active);
+    uploadedImage.value = undefined;
+}
+
+const bannersStore = useBannersStore();
+
+const onSubmit = handleSubmit(async (values) => {
+    const bannerRequest: BannerRequest = {
+        name: values['Name'],
+        duration: values['Duration'] * 1000,
+        active: values['Active'],
+        startDate: values['Start date'].toISOString(),
+        endDate: values['End date'].toISOString()
+    };
+
+    if(banner.value != undefined) {
+        // Edit existing banner
+
+        // First upload image
+        uploadedImage.value && await bannersStore.updateBannerImage(banner.value.id, uploadedImage.value);
+        // Then update and receive updated
+        const resContent = await bannersStore.updateBanner(banner.value.id, bannerRequest);
+
+        banner.value = resContent.data;
+
+        visible.value = false;
+    } else {
+        // Create new banner
+
+        // First create banner
+        const resContent = await bannersStore.createBanner(bannerRequest);
+        // Then set image
+        uploadedImage.value && await bannersStore.updateBannerImage(resContent.data.id, uploadedImage.value);
+
+        await bannersStore.fetchBanners();
+
+        visible.value = false;
+    }
+});
+
+const onImgUpload = (e: Event) => {
+    const el = (e.target as HTMLInputElement);
+    if (el == null || el.files == null) return;
+    uploadedImage.value = el.files[0]!!;
+};
+</script>
+<style>
```

Comment:
Why no inline style classes? :(

Reply:
This is more readable, also copied from primevue but adapted to work for my use case. Not the most pretty as editing products also uses this, but is an issue for later.